test/filters.js

```diff
@@ -707,6 +713,93 @@ describe('filters', function () {
+      checkForError('{{ -100 | toCurrency: "USD" }}', errorMessage));
+
+    it("should throw error when currency value is a string", () =>
+      checkForError('{{ test | toCurrency: "USD" }}', errorMessage));
+
+    it("should throw error when currency type is a number", () =>
+      checkForError("{{ 100 | toCurrency: 1 }}", errorMessage));
+
+    it("should throw error when currency type is a symbol", () =>
+      checkForError("{{ 100 | toCurrency: $ }}", errorMessage));
+  });
+
+  describe("toDuration", function(){
+
+    const durationTypeErrMsg = "duration type is incorrect";
+    const invalidDurationTypeOrValueErrMsg = "invalid duration value or type";
+
+    it("should return {value: 10, type: 'DAYS', days: 10}", () => {
+      const dst = {value: 10, type: 'DAYS', days: 10};
+      return test('{{ 10 | toDuration: "Days" }}', JSON.stringify(dst))
+    })
+
+    it("should return {value: 7, type: 'weeks', days: 49}", () => {
+      const dst = {value: 7, type: 'WEEKS', days: 49};
+      return test('{{ 7 | toDuration: "weeks" }}', JSON.stringify(dst))
+    })
+
+    it("should return {value: 5, type: 'MONTHS', days: 150}", () => {
+      const dst = {value: 5, type: 'MONTHS', days: 150};
+      return test('{{ 5 | toDuration: "months" }}', JSON.stringify(dst))
+    })
+
+    it("should return {value: 2, type: 'YEARS', days: 730}", () => {
+      const dst = {value: 2, type: 'YEARS', days: 730};
+      return test('{{ 2 | toDuration: "Years" }}', JSON.stringify(dst))
+    })
+
+    it("should throw error when duration type is not days, weeks, months or years", () =>
+      checkForError('{{ 2 | toDuration: "fortnight" }}', durationTypeErrMsg));
+
+    it("should throw error when duration value is not a number", () =>
+      checkForError('{{ "test" | toDuration: "days" }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration value is NaN", () =>
+      checkForError('{{ NaN | toDuration: "days" }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration value is negative number", () =>
+      checkForError('{{ -10 | toDuration: "days" }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration type is not a string", () =>
+    checkForError('{{ 10 | toDuration: 10 }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration value is undefined", () =>
+    checkForError('{{ | toDuration: "months" }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration value is null", () =>
+    checkForError('{{ null | toDuration: "months" }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration type is null", () =>
+    checkForError('{{ 10 | toDuration: null }}', invalidDurationTypeOrValueErrMsg));
+
+    it("should throw error when duration type is undefined", () =>
+    checkForError('{{ 10 | toDuration:  }}', invalidDurationTypeOrValueErrMsg));
+  })
+
```

Comment:
Let's also add tests where we are assigning value from an existing variable.

Ex. `{% assign durVar = existing_dur_var.value | toDuration : existing_dur_var.type%}`

Reply:
have added now.